nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/audit/ProcessGroupAuditor.java

```diff
@@ -246,6 +252,48 @@ public void enableComponentsAdvice(ProceedingJoinPoint proceedingJoinPoint, Stri
         }
 
         saveUpdateProcessGroupAction(groupId, operation);
+        if (components.get(Component.Processor) != null) {
+            saveUpdateConnectableActions(components.get(Component.Processor), operation, Component.Processor);
+        }
+        if (components.get(Component.InputPort) != null) {
+            saveUpdateConnectableActions(components.get(Component.InputPort), operation, Component.InputPort);
+        }
+        if (components.get(Component.OutputPort) != null) {
+            saveUpdateConnectableActions(components.get(Component.OutputPort), operation, Component.OutputPort);
+        }
+    }
+
+    private Map<Component, List<Connectable>> getComponents(final String groupId, final Collection<String> componentIds) {
```

Comment:
Is it necessary to build up the intermediate `Map` as opposed to iterating through the Component IDs and then adding to the list of Actions?

Reply:
Thanks @NissimShiman. I don't think there is much value in the intermediate Map, the order is a interesting note, but I don't think it should be a concern. The preference here, as an auditing method, should be on efficiency, within reason.